core/src/main/java/org/keycloak/Config.java

```diff
@@ -24,9 +24,9 @@
  */
 public class Config {
 
-    private static ConfigProvider configProvider = new SystemPropertiesConfigProvider();
+    private static volatile ConfigProvider configProvider = new SystemPropertiesConfigProvider();
 
-    public static void init(ConfigProvider configProvider) {
+    public static synchronized void init(ConfigProvider configProvider) {
```

Comment:
synchronization isn't necessary

Reply:
@shawkins Yep, marking the class var volatile might be sufficient in this context, right?

Reply:
Right